File: python/find_pivot_index.py
from typing import List


class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        leftSum, rightSum = 0, sum(nums)
        index = 0
        while index < len(nums):
            leftSum += nums[index - 1] if index - 1 >= 0 else 0
            rightSum -= nums[index]
            if leftSum == rightSum:
                return index
            index += 1
        return -1

File: src/FindPivotIndex.java
import java.util.Arrays;

public class FindPivotIndex {
    public int pivotIndex(int[] array) {
        int sum = Arrays.stream(array).sum();
        for (int index = 0, leftSum = 0 ; index < array.length ; index++) {
            leftSum += index - 1 >= 0 ? array[index - 1] : 0;
            sum -= array[index];
            if (leftSum == sum) {
                return index;
            }
        }
        return -1;
    }
}
